modules/widgets/src/view-selector-widget.tsx

```diff
@@ -0,0 +1,199 @@
+// ViewSelectorWidget.tsx
+// deck.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+
+import {h, render} from 'preact';
+import {useState, useRef, useEffect} from 'preact/hooks';
+import type {WidgetPlacement} from '@deck.gl/core';
+import type {WidgetImplProps} from './widget-impl';
+import {WidgetImpl} from './widget-impl';
+import {IconButton} from './components';
+
+/** The available view modes */
+export type ViewMode = 'single' | 'split-horizontal' | 'split-vertical';
+
+/** Properties for the ViewSelectorWidget */
+export type ViewSelectorWidgetProps = WidgetImplProps & {
+  /** Widget positioning within the view. Default 'top-left'. */
+  placement?: WidgetPlacement;
+  /** Tooltip label */
+  label?: string;
+  /** The initial view mode. Defaults to 'single'. */
+  initialViewMode?: ViewMode;
+  /** Callback invoked when the view mode changes */
+  onViewModeChange?: (mode: ViewMode) => void;
+};
+
+/**
+ * A widget that renders a popup menu for selecting a view mode.
+ * It displays a button with the current view mode icon. Clicking the button
+ * toggles a popup that shows three icons for:
+ * - Single view
+ * - Two views, split horizontally
+ * - Two views, split vertically
+ */
+export class ViewSelectorWidget extends WidgetImpl<ViewSelectorWidgetProps> {
+  static defaultProps: Required<ViewSelectorWidgetProps> = {
+    ...WidgetImpl.defaultProps,
+    id: 'view-selector-widget',
+    placement: 'top-left',
+    label: 'Split View',
+    initialViewMode: 'single',
+    onViewModeChange: () => { },
+  };
+
+  className = 'deck-widget-view-selector';
+  placement: WidgetPlacement = 'top-left';
+
+  constructor(props: ViewSelectorWidgetProps = {}) {
+    super({...ViewSelectorWidget.defaultProps, ...props});
+    this.placement = props.placement ?? this.placement;
+  }
+
+  setProps(props: Partial<ViewSelectorWidgetProps>) {
+    super.setProps(props);
+    this.placement = props.placement ?? this.placement;
+  }
+
+  onRenderHTML() {
+    const element = this.element;
+    if (!element) return;
+    // Position the widget (customize as needed)
+    element.style.position = 'absolute';
+    element.style.bottom = '10px';
+    element.style.right = '10px';
```

Comment:
Positioning the widget absolutely is likely breaking other styles. The dropdown would be absolute but the icon button should stay in the flow